Sensu go/entities

mercul3s opened the pull request:
Closes #265

calebhailey commented:
I made a quick pass through the 0.21 documentation and fixed all of the h1 headings. I'm not going to bother to go back through the old versions, but if someone else wants to that would be great!

portertech commented:
:+1:

mercul3s committed:
Added enity attribute descriptions and examples

mercul3s committed:
Completed entities reference

Added links to additional attributes in page
Cleaned up examples, added section for how entities work and 2.0
changes

mercul3s committed:
Added section for proxy entities

mercul3s committed:
Added note about entity specific subscriptions.

palourde commented on content/sensu-core/2.0/reference/entities.md:
```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
+## New and improved entities
 
-## Agent entities
+An `entity`, formally known as a `client` in Sensu 1.x, represents anything (e.g. server, container, network switch, etc) that needs to be monitored. Sensu 2.0 uses an updated data model that allows for it to acknowledge the full range of infrastructure, runtime and application types that compose a complete monitoring environment (from server hardware to "serverless" functions). Sensu no longer f ...
```
s/Entity/entity/

palourde commented on content/sensu-core/2.0/reference/entities.md:
```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
+description  | The entity type, validated with go regex [`\A[\w\.\-]+\z`](https://regex101.com/r/zo9mQU/2)
+required     | true
+type         | string 
+example      | {{< highlight shell >}}"class": "agent"{{</ highlight >}}
```
We should probably list the allowed values, `agent` & `proxy`!

grepory replied:
I don’t think there are actually allowed values here, but agent and proxy are reserved. Agent for sensu-agent processes and proxy for proxy entities.

palourde replied:
You are totally right, it's actually not enforced, my bad!

mercul3s replied:
Should I add `agent` as a default value?

palourde replied:
AFAIK the agent will set that value for its own entity and use `proxy` for result submitted through the socket & proxy checks, but otherwise it's not enforced by the API/backend. As Greg mentioned in its deleted comment, maybe the role of this attribute is not clear enough?

mercul3s replied:
Ok that makes sense. I'll clarify that this value is set by the agent/proxy.

palourde commented on content/sensu-core/2.0/reference/entities.md:
```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
+description  | A list of subscription names for the entity. 
+required     | false 
+type         | array 
+default      | The entity-specific subscription.
```
Could we give an example or the format specs, i.e., `entity:{ID}`?

mercul3s replied:
absolutely.

palourde commented on content/sensu-core/2.0/reference/entities.md:
```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
+required     | false 
+type         | string 
+example      | {{< highlight shell >}}
+  "organization": "default"
```
I just realized the spaces at the beginning of this line change the indentation of the examples for the org & env:

<img width="878" alt="screen shot 2018-04-11 at 10 20 49 am" src="https://user-images.githubusercontent.com/4105580/38622887-3c71e0d2-3d72-11e8-842d-97f3c7dd275b.png">

mercul3s replied:
🤦‍♀️  I don't even know why those had linebreaks and spaces. Dang you copy pasta!

palourde commented on content/sensu-core/2.0/reference/entities.md:
```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
-### Management
+redact       | 
+-------------|------ 
+description  | List of items to redact from log messages. 
```
We should probably specify that when using this attribute, the default values get overridden and not appended.

mercul3s replied:
Oh, that is something I wasn't aware of - will add a note.

palourde commented on content/sensu-core/2.0/reference/entities.md:
```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
-sensuctl entity delete entity.localhost
+arch         | 
+-------------|------ 
+description  | The entity's system architecture. 
```
We should probably specify that we use the binary architecture (`runtime.GOARCH`) to determine the system architecture, meaning if an `amd64` system runs the `386` binary, it will be reported as `386` in Sensu!

For reference: https://github.com/sensu/sensu-go/issues/154

mercul3s replied:
👍

palourde reviewed (changes requested):
Thank you for this tremendous work 🙏🙏 🙏 

Just some minor concerns, otherwise it's perfect!

grepory commented on content/sensu-core/2.0/reference/entities.md:
```diff
@@ -4,82 +4,355 @@ description: "The entities reference guide."
+Proxy entities (formerly known as proxy clients, "Just-in-time" or "JIT" clients) are dynamically created entities, added to the entity store if an entity does not already exist for a check result. Proxy entity registration differs from keepalive-based registration because the registration event happens while processing a check result (not a keepalive message). Sensu proxy entities allow Sensu to ...
 
-An **entity**, formally known as a client in Sensu 1.x, represents anything (e.g. server, container, network switch, etc) that needs to be monitored. It can contain system information such as the hostname, OS, platform, and version. An Entity not only provides context to event data (what/where the event is from) but an event's uniqueness is determined by the check name and the ID of the entity up ...
+## New and improved entities
```
It seems like this should come before proxy entities?

mercul3s replied:
I'm fine with moving it, but was following the convention for where "New and improved ..." falls in the current 2.0 documentation (it's after the "how x works" sections). If we move it here we might want to move it for everything else too.

palourde replied:
Yeah, I'm still not sure where this section should placed 🤔

mercul3s replied:
I think we should leave these sections later in the document to avoid confusing users new to Sensu with potentially two sets of vocabulary.

mercul3s committed:
Addressing PR feedback

mercul3s committed:
Clarify class attribute value

mercul3s commented:
@palourde feedback addressed!

palourde reviewed (approved):
![](https://user-images.githubusercontent.com/4105580/38635219-b367be12-3d92-11e8-8adf-907d5b667c0c.gif)
